src/sgame/sg_bot_ai.cpp

```diff
@@ -813,7 +836,12 @@ AINodeStatus_t BotActionFight( gentity_t *self, AIGenericNode_t *node )
 		return STATUS_SUCCESS;
 	}
 
-	if ( !mind->nav().havePath )
+	if ( !mind->nav().havePath && !mind->hasOffmeshGoal )
+	{
+		return STATUS_FAILURE;
+	}
+
+	if ( mind->hasOffmeshGoal && !BotTargetIsVisible( self, self->botMind->goal, MASK_OPAQUE ) )
```

Comment:
Why is MASK_OPAQUE used here and MASK_SHOT in the other place?

Reply:
I'm not entirely sure. I went for MASK_OPAQUE first, because this is used nearby. But it turns out that this will make bots fire at their own buildings if these are in the line of sight. I forgot to change this occurance.